charts/k8s-infra/templates/_config.tpl

```diff
@@ -343,20 +343,30 @@ processors:
 {{- define "opentelemetry-collector.resourceDetectionConfig" -}}
 processors:
   resourcedetection:
+    # detectors: include ec2/eks for AWS, gcp for GCP and azure/aks for Azure
+    # env detector included below adds custom labels using OTEL_RESOURCE_ATTRIBUTES envvar (set envResourceAttributes value)
     detectors:
-      {{- toYaml .Values.presets.resourceDetection.detectors | nindent 6 }}
-    {{- if has "k8snode" .Values.presets.resourceDetection.detectors }}
+      - env
+      - k8snode
+      {{- if eq "aws" .Values.global.cloud }}
+      - eks
+      - ec2
+      {{- end }}
+      {{- if hasPrefix "gcp" .Values.global.cloud }}
+      - gcp
```

Comment:
Why `hasPrefix` for gcp and `eq` for others?

Reply:
- `gcp` cloud for GKE standard
- `gcp/autopilot` for GKE Autopilot